src/material-experimental/mdc-snack-bar/snack-bar-container.ts

```diff
@@ -100,25 +104,38 @@ export class MatSnackBarContainer extends BasePortalOutlet
    */
   @ViewChild('label', {static: true}) _label: ElementRef;
 
+  /**
+   * The div container that is hidden from screen readers and initially holds to snack bar
+   * content.
+   */
+  @ViewChild('nonLive') _nonLiveElementRef: ElementRef<HTMLElement>;
+
+  /** The div container that is a live region and will have the snack bar content move into. */
+  @ViewChild('live') _liveElementRef: ElementRef<HTMLElement>;
+
   constructor(
       private _elementRef: ElementRef<HTMLElement>,
       public snackBarConfig: MatSnackBarConfig,
       private _platform: Platform) {
     super();
 
-    // Based on the ARIA spec, `alert` and `status` roles have an
-    // implicit `assertive` and `polite` politeness respectively.
+    // Use aria-live rather than a live role like 'alert' or 'status'
+    // because NVDA and JAWS have show inconsitent behavior with live roles.
     if (snackBarConfig.politeness === 'assertive' && !snackBarConfig.announcementMessage) {
-      this._role = 'alert';
+      this._live = 'assertive';
     } else if (snackBarConfig.politeness === 'off') {
-      this._role = null;
+      this._live = 'off';
     } else {
-      this._role = 'status';
+      this._live = 'polite';
     }
 
     // `MatSnackBar` will use the config's timeout to determine when the snack bar should be closed.
     // Set this to `-1` to mark it as indefinitely open so that MDC does not close itself.
     this._mdcFoundation.setTimeoutMs(-1);
+
+    // Once the snack bar's entrance animation ends do the DOM manipulation to announce the
+    // message.
+    this._onEnter.pipe(take(1)).subscribe(() => this._screenReaderAnnounce());
```

Comment:
I was first using `setTimeout` with a 100ms delay, but switched to using `_onEnter` to avoid creating another timer. After testing with animations turned off, I found that would cause issues with NVDA/JAWS. Switched back to using `setTimeout` in `_screenReaderAnnounce` and calling the method in `enter`.